Improvements to ts->xlsx->ts

s-t-e-f-a-n opened the pull request:
Improved qTsConverter to produce ts-A->xlsx->ts-B with ts-B identical to ts-A and taking into account following improvements:
- included source and sourcelanguage information
- includes xlsx headers and support for: Context, ID, Source, Translation, TranslationType, comment, extracomment, translatorcomment, Location
- added scripts for bulk conversion of ts-files and xlsx-files
- added script to build on Debian / Ubuntu for usage as a cli tool

Commit:
compile-cli.sh angelegt

s-t-e-f-a-n committed:
Extended by HeaderObject

s-t-e-f-a-n committed:
Added support: translationtype, comment, extracomment, translatorcomment - all optional
Checked for: ts->xlsx->ts
Todo: cleaned up

s-t-e-f-a-n committed:
Cleanup

s-t-e-f-a-n committed:
Added scripts to process ts->xlsx and xlsx->ts as batched folder

s-t-e-f-a-n committed:
Added support for (optional) id-attribute of <message> in ts

guerinoni commented on src/internal/TitleHeaders.hpp:
```diff
@@ -1,9 +1,41 @@
+    </message>
+</context>
+</TS>
+ */
```
Since is in README you can remove it from here

guerinoni reviewed (commented):
Please remove `.vscode` stuff and add a section in readme for documenting all 3 new scripts :)

s-t-e-f-a-n committed:
aligned tst_ts2csv

s-t-e-f-a-n committed:
Tests tst_csv2ts, tst_ts2csv, ts2_xlx run smooth; tst_xlsx needs fixes

s-t-e-f-a-n committed:
Test tst_xlsx2ts up and running
Added description for scripts compile-cli.sh in readme
Added description for scripts ts-to-xlsx.sh and xlsx-to-ts.sh in readme both for stack conversion of folder content
Removed .vscode stuff

s-t-e-f-a-n committed:
clang-format applied

s-t-e-f-a-n committed:
Removed dead code based on QDomDocument
Removed qSetGlobalQHashSeed which was introduced in DomDocument handling for getting a reproducable XML attribute order (did not work)
